mqtt/mqtt.js

```diff
@@ -1,211 +1,139 @@
     server.listen(config.ws_port);
     return server;
 }
-/**
- * MQTT
- */
+
 function startMqtt() {
     return net.createServer(aedes.handle).listen(config.mqtt_port);
 }
-/**
- * NATS
- */
-nats.subscribe('channel.*', function (msg) {
-
-    var m = message.RawMessage.decode(Buffer.from(msg))
 
-    if (m.Protocol == 'mqtt') {
-        // Ignore MQTT loopback,
-        // packet has been already published by MQTT broker
-        // before sending it to NATS
-        return;
-    }
-
-    // Parse and adjust content-type
-    if (m.ContentType == "application/senml+json") {
-        m.ContentType = "senml-json"
-    }
-
-    var packet = {
-        cmd: 'publish',
-        qos: 2,
-        topic: 'mainflux/channels/' + m.Channel + '/messages/' + m.ContentType,
-        payload: m.Payload,
-        retain: false
-    };
+nats.subscribe('channel.*', function (msg) {
+    var m = message.RawMessage.decode(Buffer.from(msg)),
+        packet = {
+            cmd: 'publish',
+            qos: 2,
+            topic: 'channels/' + m.Channel + '/messages',
+            payload: m.Payload,
+            retain: false
+        };
 
     aedes.publish(packet);
 });
 
-/**
- * Hooks
- */
-// AuthZ PUB
-aedes.authorizePublish = function (client, packet, callback) {
-    // Topics are in the form `mainflux/channels/<channel_id>/messages/senml-json`
-    var channel = packet.topic.split('/')[2];
-
-    /**
-     * Check if PUB is authorized
-     */
-    var options = {
-        url: config.auth_url + ':' + config.auth_port + '/channels/' + channel + '/access-grant',
-        method: 'GET',
-        headers: {
-            'Content-Type': 'application/json',
-            'Authorization': client.password
-        }
-    };
-
-    request(options, function (err, res) {
-        var error = null;
-        var msg = {};
-        if (res && (res.statusCode === 200)) {
-            console.log('Publish authorized OK');
-
-          var msg = message.RawMessage.encode({
-              /**
-               * We must publish on NATS here, because on_publish() is also called
-               * when we receive message from NATS from other adapters (in nats.subscribe()),
-               * so we must avoid re-publishing on NATS what came from other adapters
-               */
-              Publisher: client.id,
-              Channel: channel,
-              Protocol: 'mqtt',
-              ContentType: packet.topic.split('/')[4],
-              Payload: packet.payload
+aedes.authorizePublish = function (client, packet, publish) {
+    // Topics are in the form `channels/<channel_id>/messages`
+    var channel = packet.topic.split('/')[1];
+
+    things.CanAccess({
+        token: client.password,
+        chanID: channel
+    }, onAuthorize);
+
+    function onAuthorize(err, res) {
+        if (!err) {
+            logger.info('authorized publish');
+            
+            var rawMsg = message.RawMessage.encode({
```

Comment:
Since variables in javascript are function-scoped, not block-scoped, it makes sense to declare (or define) all locals at the beginning of the function (in this case only declare).